### ISSUE TITLE
External DNS: support for SRV records

### ISSUE THREAD

**Issue by raphink** [rancher/rancher#3642]
It would be useful (e.g. for prometheus) to have external-dns support SRV records in addition to A records.

**Comment by mcanevet** [comment]
+1

**Comment by AlexRRR** [comment]
There is already discussion about this in the Rancher forums, but just in case this does get traction. +1

**Comment by raphink** [comment]
I had no idea there were rancher forums. Do you have a link to the thread?

**Comment by vincent99** [comment]
The only post I can think of/find involving SRV records is https://forums.rancher.com/t/dns-caching-issues/1566/18 , but that was for the (internal) rancher-dns, not external-dns.  Anyway issues and work being done to address them would be here, not in the forum.

**Comment by raphink** [comment]
It would definitely also be useful in internal DNS resolutions, but this bug is specifically about external-dns.

**Comment by AlexRRR** [comment]
Yes, I mixed up the rancher forums and some thread I read somewhere else ( [https://github.com/prometheus/prometheus/issues/1203](here)), I've been spending some time in the forums, and thought I had seen it there.
